### PR TITLE
Added timings information, updated version of statick to use.

### PR DESCRIPTION
**Pull request by gregtkogut**
This initial PR is draft, and needs a new 0.9.x release of statick before the new 'timings' option is useable.  This PR is draft for staging only.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
-FROM ghcr.io/sscpac/statick:0.7.2
+FROM ghcr.io/sscpac/statick:0.8.1
 
 LABEL "name"="Statick Action"
-LABEL "version"="0.7.2"
+LABEL "version"="0.8.1"
```
Comment (denewiler) [inline Dockerfile:4]: If this is going to support the `--timings` flag should the version go to `0.9.0`?

**Comment by denewiler** [comment]
Looks like the README needs an entry for the `--timings` flag.

File: action.yml [hunk 2]
```diff
@@ -29,6 +29,10 @@ inputs:
     description: "Profile configuration"
     required: false
     default: profile.yaml
+  timings:
```
Comment (xydesa) [inline action.yml:32]: would it be good to add the new `level` flag as well?

Comment (gregtkogut) [reply action.yml:36]: df43ff6

File: Dockerfile (hunk 1 above)
Comment (gregtkogut) [reply Dockerfile]: df43ff6    Still needs testing, as 0.9.0 has not be published yet.

File: action.yml (hunk 2 above)
Comment (gregtkogut) [reply action.yml:36]: Still needs testing as this is not testable until 0.9.0 image is released.

File: README.md [hunk 3]
```diff
@@ -40,6 +40,10 @@ Where Statick should look for files to perform static analysis on. Default: `.`
 
 The Statick profile file to use. Default: `profile.yaml`.
 
+### `timings`
+
+Output timing information to stdout. Default `false`.
+
```
Comment (denewiler) [inline README.md:46]:
```suggestion

### `level`

Scan all packages using this level. Overrides all levels set from the profile.

```

**Comment by gregtkogut** [comment]
@xydesa  or @denewiler, I believe the self-check Action is failing here because the current release of statick-md uses the old string method for the default config file.  Can we do a release of statick-md (and maybe statick-tooling?) concurrently with 0.9.0 of statick proper?

File: .github/workflows/self-test.yml [hunk 4]
```diff
@@ -14,18 +14,19 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Statick
-        uses: sscpac/statick-action@v0.0.2
+        uses: gregtkogut/statick-action@add-timings-information
```
Comment (gregtkogut) [inline .github/workflows/self-test.yml:17]: - [ ] Change this line, and the below two similar lines to the expected release version before merging.

**Comment by xydesa** [comment]
@gregtkogut statick-tooling hasn't been updated since it's last release, does it's config file need to be updated first?
We can do a new release of statick-md

**Comment by xydesa** [comment]
all of the packages here will need new releases probably: https://github.com/sscpac/statick/blob/main/docker/Dockerfile#L76
and then we'll need a new docker image too i think.

**Comment by gregtkogut** [comment]
> all of the packages here will need new releases probably: https://github.com/sscpac/statick/blob/main/docker/Dockerfile#L76 and then we'll need a new docker image too i think.

Let me know if I should try making the releases, or if you or @denewiler  would like to.  I think you guys would be more aware of any potential side effects of a new release.

**Comment by tdenewiler** [comment]
> > all of the packages here will need new releases probably: https://github.com/sscpac/statick/blob/main/docker/Dockerfile#L76 and then we'll need a new docker image too i think.
> 
> Let me know if I should try making the releases, or if you or @denewiler would like to. I think you guys would be more aware of any potential side effects of a new release.

New releases have been made for statick, statick-md, statick-planning, and statick-tex. There are PRs open to update the version and changelog for the final two packages: statick-tooling and statick-web. When those PRs are approved/merged I can help getting new releases created/published.

**Comment by denewiler** [comment]
> > > all of the packages here will need new releases probably: https://github.com/sscpac/statick/blob/main/docker/Dockerfile#L76 and then we'll need a new docker image too i think.
> > 
> > 
> > Let me know if I should try making the releases, or if you or @denewiler would like to. I think you guys would be more aware of any potential side effects of a new release.
> 
> New releases have been made for statick, statick-md, statick-planning, and statick-tex. There are PRs open to update the version and changelog for the final two packages: statick-tooling and statick-web. When those PRs are approved/merged I can help getting new releases created/published.

Final PRs are merged and releases published for statick-tooling and statick-web. I'll keep an eye on this to see if there is anything else we need to update for this release.

**Comment by gregtkogut** [comment]
@tdenewiler , @xydesa , all checks are finally passing after I built and pushed a new statick docker image.   Thanks for that work.  This is ready for review of the actual functionality.

**Review by xydesa — APPROVED** [review]

**Comment by tdenewiler** [comment]
I am testing the new statick-action with the statick-tex plugin repository. I am seeing discrepancies between using the proposed statick-action and running Statick locally. The proprosed statick-action has pylint warnings that are not showing up locally.

Digging a little deeper I found that the Docker image used by statick-action has pylint version 2.4.4 and locally I have pylint version 2.15.5. @xydesa saw that the Dockerfile used to create the image used by statick-action is installing pylint from apt, whereas our other uses of Statick install pylint from PyPI (pip). There may be other dependencies that could move from apt to pip install, such as pylint-django.

**Comment by tdenewiler** [comment]
Looks like we need a new Statick release then we can update this PR. The new release is needed so that we get the Python tools from PyPI rather than apt, which is a recent change in the statick repo.

**Comment by tdenewiler** [comment]
Looks like we need a new Statick release then we can update this PR. The new release is needed so that we get the Python tools from PyPI rather than apt, which is a recent change in the statick repo.

PR for new release is at sscpac/statick#456.

**Review by tdenewiler — APPROVED** [review]

**Review by denewiler — APPROVED** [review]

**Pull request merged**